pkg/config/config.go

```diff
@@ -716,6 +716,10 @@ func InitConfig(config Config) {
 	config.BindEnvAndSetDefault("cluster_agent.max_leader_connections", 100)
 	config.BindEnvAndSetDefault("cluster_agent.client_reconnect_period_seconds", 1200)
 	config.BindEnvAndSetDefault("cluster_agent.collect_kubernetes_tags", false)
+	config.BindEnvAndSetDefault("cluster_agent.kubernetes_resources_collection.pod_annotations_exclude", []string{
+		"kubectl.kubernetes.io/last-applied-configuration",
+		"^ad.datadoghq.com\\/([[:alnum:]].*.)?(checks|check_names|init_configs|instances)$",
```

Comment:
Dots are matching any character. In order to match a dot, it needs to be escaped.

```suggestion
		`^kubectl\.kubernetes\.io/last-applied-configuration$`,
		`^ad\.datadoghq\.com/([[:alnum:]]+\.)?(checks|check_names|init_configs|instances)$`,
```